tests/mock_module/resource.py

```diff
@@ -12,14 +12,16 @@ class ResourceConfig(RecordResourceConfig):
     list_route = "/mocks"
     item_route = f"{list_route}/<pid_value>"
 
+    # IDEA: Formalize the concept of thunk in our codebase and use it
+    # TODO: Changing list_route / item_route shouldn't require redefining these
```

Comment:
I disagree with this point. I'd like to have links generation and app routing completely separated. This makes them decoupled components which are far easier to maintain.

Reply:
Not sure I understand this one :) . App routing and link generation are already coupled: routing uses the `list_route`. And even if they were not "code" coupled, they are semantically coupled (not a bad coupling): we should be generating links to the endpoints defined via `iist_route` / `item_route`...

Reply:
discussed IVL -- we are not coupling the two - if ui link changes, it's up to developer to also change it in the links_config